### PR TITLE
Added support of segmentation mask in RandomShear Layer

### PR DESCRIPTION
**Pull request by cosmo3769**
# What does this PR do?

Related #1992 

Add support of segmentation mask in Augmix layer. Here is the [colab link](https://colab.research.google.com/drive/1lBOl5ruK7AhZf9YOh4v3c9F1oYq2kob-?usp=sharing) for demo.

Output generated by demo:

Screenshot 2023-08-11 at 8 23 29 PM

## Who can review?

@ianstenbit

### CODE REVIEW

File: keras_cv/layers/preprocessing/random_shear.py [hunk 1]
```diff
@@ -219,6 +219,33 @@ def _build_shear_y_transform_matrix(shear_y):
...
+        self, segmentation_masks, transformations, **kwargs
+    ):
+        x, y = transformations["shear_x"], transformations["shear_y"]
+
+        if x is not None:
+            transforms_x = self._build_shear_x_transform_matrix(x)
+            segmentation_masks = preprocessing.transform(
+                images=segmentation_masks,
+                transforms=transforms_x,
+                interpolation=self.interpolation,
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/random_shear.py:232]: We should use `nearest` interpolation for segmentation masks

**Review by ianstenbit — APPROVED** [review]
Looks good to me, just one small fix!

Thank you for the PR!

File: keras_cv/layers/preprocessing/random_shear.py (hunk 1 above)
Comment (cosmo3769) [reply keras_cv/layers/preprocessing/random_shear.py]: Done.

**Review by ianstenbit — APPROVED** [review]

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Looks like we missed one more little thing -- thanks!

File: keras_cv/layers/preprocessing/random_shear.py [hunk 2]
```diff
@@ -219,6 +219,33 @@ def _build_shear_y_transform_matrix(shear_y):
...
+                fill_mode=self.fill_mode,
+                fill_value=self.fill_value,
+            )
+
+        if y is not None:
+            transforms_y = self._build_shear_y_transform_matrix(y)
+            segmentation_masks = preprocessing.transform(
+                images=segmentation_masks,
+                transforms=transforms_y,
+                interpolation=self.interpolation,
```
[the remark this answers is not in the document]
Comment (cosmo3769) [reply keras_cv/layers/preprocessing/random_shear.py]: Apologies. I missed that. 😅

**Review by ianstenbit — APPROVED** [review]
Awesome -- thank you!

**Pull request merged**